### DISCUSSION TITLE
How to use the DynamicAABBTreeCollisionManager in C++

### DISCUSSION THREAD

**Discussion by lwygzh** [coal-library/coal#532] [Q&A] [answered]
Hi,

I was trying out the C++ API, and I am a bit confused with how the DynamicAABBTreeCollisionManager should be used. I would appreciate any suggestions to make the my code more efficient. Currently this is what I am doing

```cpp
hpp::fcl::DynamicAABBTreeCollisionManager m = hpp::fcl::DynamicAABBTreeCollisionManager();

// load objects
std::vector<hpp::fcl::CollisionObject*> collisionObjects;
collisionObjects.push_back(ObjPtr0.get());
collisionObjects.push_back(ObjPtr1.get());
collisionObjects.push_back(ObjPtr2.get());

m.registerObjects(collisionObjects);

hpp::fcl::CollisionGeometryPtr_t modelPtr = std::make_shared<hpp::fcl::Box>(0.5, 2.4, 2.0);
std::shared_ptr<hpp::fcl::CollisionObject> objPtr = std::make_shared<hpp::fcl::CollisionObject>(modelPtr);

// I am skipping the transformation settings for now

hpp::fcl::CollisionCallBackDefault callback;
callback.init();
callback.data.result.clear();
callback.data.done = false;

m.collide(objPtr.get(), &callback);
bool isCollision = callback.data.result.isCollision();
```

I think this works, but I am not sure if this is the intended usage, or if I am doing anything wrong. The tests might be doing something similar but I am new to C++ and the code is a bit confusing for me.

Thanks for any suggestions!

**Comment by jcarpent** [comment, marked as answer]
This looks good to me.

**Reply by lwygzh** [reply]
@jcarpent Thanks for the response! That's good to know! 

Another question I have is that it seems like if I want to use `hpp-fcl` with `pinocchio` I still somehow have to manually compute the forward kinematics and set the position and orientation of each link. I was wondering if there is exists a functionality that already achieves this I can directly use? I can also create a new discussion regarding this if you think it is suitable.
